sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/DLPDeidentifyText.java

```diff
@@ -0,0 +1,282 @@
+
+    /** @param deidentifyTemplateName Template name for data deidentification. */
+    public abstract Builder setDeidentifyTemplateName(String deidentifyTemplateName);
+
+    /**
+     * @param inspectConfig Configuration object for data inspection. If present, supersedes the
+     *     template settings.
+     */
+    public abstract Builder setInspectConfig(InspectConfig inspectConfig);
+
+    /**
+     * @param deidentifyConfig Configuration object for data deidentification. If present,
+     *     supersedes the template settings.
+     */
+    public abstract Builder setDeidentifyConfig(DeidentifyConfig deidentifyConfig);
+
+    abstract DLPDeidentifyText autoBuild();
+
+    public DLPDeidentifyText build() {
+      DLPDeidentifyText dlpDeidentifyText = autoBuild();
+      if (dlpDeidentifyText.getDeidentifyConfig() == null
+          && dlpDeidentifyText.getDeidentifyTemplateName() == null) {
+        throw new IllegalArgumentException(
+            "Either deidentifyConfig or deidentifyTemplateName need to be set!");
+      }
+      if (dlpDeidentifyText.getBatchSizeBytes() > DLP_PAYLOAD_LIMIT_BYTES) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Batch size is too large! It should be smaller or equal than %d.",
+                DLP_PAYLOAD_LIMIT_BYTES));
+      }
+      if (dlpDeidentifyText.getColumnDelimiter() == null
+          && dlpDeidentifyText.getHeaderColumns() != null) {
+        throw new IllegalArgumentException(
+            "Column delimiter should be set if headers are present.");
+      }
+      if (dlpDeidentifyText.getHeaderColumns() == null
+          && dlpDeidentifyText.getColumnDelimiter() != null) {
+        throw new IllegalArgumentException(
+            "Column headers should be supplied when delimiter is present.");
+      }
+      return dlpDeidentifyText;
+    }
+  }
+
+  public static DLPDeidentifyText.Builder newBuilder() {
+    return new AutoValue_DLPDeidentifyText.Builder();
+  }
+
+  /**
+   * The transform converts the contents of input PCollection into {@link Table.Row}s and then calls
+   * Cloud DLP service to perform the deidentification according to provided settings.
+   *
+   * @param input input PCollection
+   * @return PCollection after transformations
+   */
+  @Override
+  public PCollection<KV<String, DeidentifyContentResponse>> expand(
+      PCollection<KV<String, String>> input) {
+    return input
+        .apply(ParDo.of(new MapStringToDlpRow(getColumnDelimiter())))
+        .apply("Batch Contents", ParDo.of(new BatchRequestForDLP(getBatchSizeBytes())))
+        .apply(
+            "DLPDeidentify",
+            ParDo.of(
+                new DeidentifyText(
+                    getProjectId(),
+                    getInspectTemplateName(),
+                    getDeidentifyTemplateName(),
+                    getInspectConfig(),
+                    getDeidentifyConfig(),
+                    getHeaderColumns())));
+  }
+
```

Comment:
@mwalenia Hi Michal- I have tried with 2.23 snapshot yesterday. Do you think it was missed to pass on the side input in this step? like .withSideInputs(headerColumns)

Reply:
Thanks for your keen eye, it must have slipped our minds during review to check for the side input here. I'll try to get this fixed and ready for 2.23.

Reply:
Thank you.